lib/docker.js

```diff
@@ -51,7 +51,8 @@ function runImage(imageId, cwd, command, mount) {
   var mountDir = crossPlatformCwd(cwd);
   var mountComponent = mount ? `-v "${mountDir}:/app/src"` : '';
   var envArgComponent = directory.getFormattedEnvArgComponent(cwd);
-  var runCommand = `run -w /app/src -p 3000:3000 --rm -it ${mountComponent} ${envArgComponent} ${imageId} sh -c "${command}" || true`;
+  var commandWithPort = command.replace("$PORT", 3000).replace("${PORT}, 3000")
```

Comment:
Ah, so trying to do something like:

```
web: foo $PORT
```

?

If so, I wonder if it's an escaping issue - like `$PORT` being evaluated before it's used in the container. Could you try replacing $ with \$ instead, to see if that solves the general case? If not, I'll go ahead and merge as-is. The plan for the next iteration is to use the Docker API instead of shelling out, so I don't mind a few temporary shell hacks for now.